### DISCUSSION TITLE
Pulling Period of Report / Reporting Period for filings

### DISCUSSION THREAD

**Discussion by gweifu** [dgunning/edgartools#180] [Q&A] [answered]
Hey everyone, I'm very new to all of this, so apologies in advance if this is a dumb question...

I'm attempting to use Python and edgartools to pull certain information by accession number. My current script is as follows:
----------
from edgar import *
set_identity("John Smith johnsmith@gmail.com")

filing = get_by_accession_number("0000895421-24-000436")

print(filing)
----------
The script returns the form, filing_date, company, cik, and accession_no for the filing. This contains most of the information I'm seeking, but it is does not contain the Period of Report / Reporting period.

In this example, the filing_date is 2024-08-16 and the period of report is 2024-03-31. The period of report can be identified using filing.header, filing.attachments, filing.obj, and filing.full_text_submission, but I have been unable to find a way to extract the date at scale.

Thanks for the help!

**Comment by dgunning** [comment, marked as answer]
The `get_filings` function uses the main filing indexes which only contain `form`,` filing_date`, `company`, `cik`, and `accession_no` for the filing. 

The company filings data contains the reportDate so if you want to get the report date for filing for a give company first get the company then use that data.

```python
c = Company(895421)
df = c.filings.to_pandas().query("reportDate !=''")
df
```

I'm assuming that by 'at scale' you need a performant way to get thousands of filings with report date. It might be faster to go through the companies especially if you use local storage and download company data.

**Reply by gweifu** [reply]
Makes sense, thank you! I appreciate the quick response!
